include/eve/detail/function/simd/riscv/bit_cast.hpp

```diff
@@ -0,0 +1,586 @@
+                          as<wide<U, M>> const                              &to_as) noexcept
+requires rvv_abi<abi_t<T, N>> && rvv_abi<abi_t<U, M>>
+{
+  return bit_cast(get<0>(x), to_as);
+}
+
+template<scalar_value T, typename N, scalar_value U, typename M>
+EVE_FORCEINLINE auto
+bit_cast_(EVE_SUPPORTS(rvv_),
+          kumi::tuple<logical<eve::wide<T, N>>> const& x,
+          as<logical<wide<U, M>>> const              & to_as) noexcept
+requires rvv_abi<abi_t<T, N>> && rvv_abi<abi_t<U, M>>
+{
+  return bit_cast(get<0>(x), to_as);
+}
+
+template<scalar_value T, typename N, scalar_value U, typename M>
+EVE_FORCEINLINE wide<U, M>
+bit_cast_(EVE_SUPPORTS(rvv_), wide<T, N> const& x, as<wide<U, M>> const& to_as) noexcept
+requires rvv_abi<abi_t<T, N>> && rvv_abi<abi_t<U, M>> && same_wide_size<T, N, U, M>
+         && same_raw_wide_size<T, N, U, M>
+{
+  using in_wide  = wide<T, N>;
+  using out_wide = wide<U, M>;
+
+  constexpr auto in_c  = categorize<in_wide>();
+  constexpr auto out_c = categorize<out_wide>();
+
+  if constexpr( match(in_c, category::float_) )
+  {
+    // static_assert(!match(out_c, category::float_));
+    //  float. We need cast to the to intype with the same width.
+    using sign            = default_as_integer_sign_t<U>;
+    using out_part_scalar = as_integer_t<T, sign>;
+    using out_part_wide   = wide<out_part_scalar, N>;
+    auto part_done        = bit_cast(x, as<out_part_wide> {});
+
+    return bit_cast(part_done, to_as);
+  }
+  else
+  {
+    // first change sew, then cast to type
+    using sign            = default_as_integer_sign_t<T>;
+    using out_part_scalar = as_integer_t<U, sign>;
+    using out_part_wide   = wide<out_part_scalar, M>;
+    auto part_done        = bit_cast(x, as<out_part_wide> {});
+    return bit_cast(part_done, to_as);
+  }
+}
+template<scalar_value T, typename N, scalar_value U, typename M>
+EVE_FORCEINLINE wide<U, M>
+bit_cast_(EVE_SUPPORTS(rvv_), wide<T, N> const& x, as<wide<U, M>> const& to_as) noexcept
+requires rvv_abi<abi_t<T, N>> && rvv_abi<abi_t<U, M>> && same_wide_size<T, N, U, M>
+         && (sizeof(T) * N::value < sizeof(U) * M::value)
+{
+  if constexpr( is_aggregated_v<abi_t<T, N>> || is_aggregated_v<abi_t<U, M>> ) static_assert(false);
+  if constexpr( is_aggregated_v<abi_t<T, typename N::combined_type>>
+                || is_aggregated_v<abi_t<U, M>> )
+    static_assert(false);
+  auto extended_in = riscv_lmul_ext(x);
+  for( auto id = N::value; id < N::combined_type::value; ++id )
+    extended_in.set(id, static_cast<T>(0));
+  return bit_cast(extended_in, to_as);
+}
+template<scalar_value T, typename N, scalar_value U, typename M>
+EVE_FORCEINLINE wide<U, M>
+bit_cast_(EVE_SUPPORTS(rvv_), wide<T, N> const& x, as<wide<U, M>> const& to_as) noexcept
+requires rvv_abi<abi_t<T, N>> && rvv_abi<abi_t<U, M>> && same_wide_size<T, N, U, M>
+         && (sizeof(T) * N::value > sizeof(U) * M::value)
```

Comment:
You can only bit_cast if sizeof(T) * N::value == sizeof(U) * M::value - this overload is invalid

Reply:
Removed